agent/api/task/task.go

```diff
@@ -1241,6 +1255,44 @@ func (task *Task) shouldOverrideIPCMode(container *apicontainer.Container, docke
 	}
 }
 
+
+func (task *Task) initializeContainerOrderingForVolumes() error{
+	for _, container := range task.Containers {
+		if len(container.VolumesFrom) > 0 {
+			for _, volume := range container.VolumesFrom {
+				if _, ok := task.ContainerByName(volume.SourceContainer); !ok {
+					return fmt.Errorf("could not find container with name %s", volume.SourceContainer)
+				} else {
+					dependOn := apicontainer.DependsOn{Container: volume.SourceContainer, Condition: ContainerOrderingStartCondition}
+					container.DependsOn = append(container.DependsOn, dependOn);
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (task *Task) initializeContainerOrderingForLinks() error{
+	for _, container := range task.Containers {
+  		if len(container.Links) > 0 {
+  			for _, link := range container.Links {
+				linkParts := strings.Split(link, ":")
+				if len(linkParts) > 2 {
+					return fmt.Errorf("Invalid link format")
+				}
+				linkName := linkParts[0]
+				if _, ok := task.ContainerByName(linkName); !ok {
+					return fmt.Errorf("could not find container with name %s", linkName)
+				} else {
```

Comment:
nit: same as above, doesn't seem like we need the else branch here.

Reply:
Thanks for pointing out. Removed the else.